packages/hadron-document/src/element.ts

```diff
@@ -753,7 +756,7 @@ export class Element extends EventEmitter {
         element.collapse();
       }
     }
-    this.emit(ElementEvents.Collapsed, this);
+    this._bubbleUp(ElementEvents.Collapsed, this);
```

Comment:
We can bubble it, but then we are also create an interface that requires you to know the internal implementation. I feel like we should rather emit "visible changed" in addition to this even if appropriate

Reply:
yea I was in a fix about wether to emit VisibleElementsChanged for all the actions that could possibly change the number of visible elements or wether to listen for those action events directly in the needed place. I opted [for latter](https://github.com/mongodb-js/compass/pull/6010/files#diff-dd307ccd5a00c28fdaa74fe18ee0fe7a3595235e5a1b11c534008d54881ecc83R82-R87), it probably was not the best idea. I will adapt the suggestion here as I have better understanding now. Thank you :)

Reply:
addressed this in - [628c98d](https://github.com/mongodb-js/compass/pull/6009/commits/628c98d58fd09bb664238fe9fe0989b4fc0992ab)